src/exif.c

```diff
@@ -0,0 +1,133 @@
+            return AVIF_RESULT_INVALID_EXIF_PAYLOAD;
+        }
+        for (uint16_t field = 0; field < fieldCount; ++fieldCount) { // for each interoperability array
+            uint16_t tag;
+            uint16_t type;
+            uint32_t count;
+            uint16_t firstHalfOfValueOffset;
+            uint16_t secondHalfOfValueOffset;
+            if (!avifROStreamReadU16Endianness(&stream, &tag, isLittleEndian) ||
+                !avifROStreamReadU16Endianness(&stream, &type, isLittleEndian) ||
+                !avifROStreamReadU32Endianness(&stream, &count, isLittleEndian) ||
+                !avifROStreamReadU16Endianness(&stream, &firstHalfOfValueOffset, isLittleEndian) ||
+                !avifROStreamReadU16Endianness(&stream, &secondHalfOfValueOffset, isLittleEndian)) {
+                return AVIF_RESULT_INVALID_EXIF_PAYLOAD;
+            }
+            // Orientation attribute according to JEITA CP-3451C section 4.6.4 (TIFF Rev. 6.0 Attribute Information):
+            if (tag == 0x0112 && type == /*SHORT=*/0x03 && count == 0x01) { // Exif IFD
+                // Mapping from Exif orientation as defined in JEITA CP-3451C section 4.6.4.A Orientation
+                // to irot and imir boxes as defined in HEIF ISO/IEC 28002-12:2021 sections 6.5.10 and 6.5.12.
+                switch (firstHalfOfValueOffset) {
+                    case 1: // The 0th row is at the visual top of the image, and the 0th column is the visual left-hand side.
+                        *flags = AVIF_TRANSFORM_NONE;
+                        irot->angle = 0; // ignored
+                        imir->mode = 0;  // ignored
+                        return AVIF_RESULT_OK;
+                    case 2: // The 0th row is at the visual top of the image, and the 0th column is the visual right-hand side.
+                        *flags = AVIF_TRANSFORM_IMIR;
+                        irot->angle = 0; // ignored
+                        imir->mode = 1;
+                        return AVIF_RESULT_OK;
+                    case 3: // The 0th row is at the visual bottom of the image, and the 0th column is the visual right-hand side.
+                        *flags = AVIF_TRANSFORM_IROT;
+                        irot->angle = 2;
+                        imir->mode = 0; // ignored
+                        return AVIF_RESULT_OK;
+                    case 4: // The 0th row is at the visual bottom of the image, and the 0th column is the visual left-hand side.
+                        *flags = AVIF_TRANSFORM_IMIR;
+                        irot->angle = 0; // ignored
+                        imir->mode = 0;
+                        return AVIF_RESULT_OK;
+                    case 5: // The 0th row is the visual left-hand side of the image, and the 0th column is the visual top.
+                        *flags = AVIF_TRANSFORM_IROT | AVIF_TRANSFORM_IMIR;
+                        irot->angle = 1; // applied before imir according to Chrome's rendering behavior
```

Comment:
I believe the order in which `irot` and `imir` are applied is specified by HEIF. Please check HEIF and verify that. If that's the case, this comment should say "according to HEIF", not "according to Chrome's rendering behavior".

Reply:
I already searched for it and failed to find an explicit proof in HEIF. However I just checked MIAF again and it is specified there in section 7.3.6.7; I think I missed it because "irot" is not part of MIAF.